CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+
+project(KokkosTools CXX)
+
+# Set policies
+cmake_policy(SET CMP0054 NEW) # simplify if()
+cmake_policy(SET CMP0057 NEW) # support IN_LIST in if()
+cmake_policy(SET CMP0111 NEW) # error if library not found
+
+# Disable in-source builds to prevent source tree corruption.
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
+  message(FATAL_ERROR "FATAL: In-source builds are not allowed. You should create a separate directory for build files.")
+endif()
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+message(STATUS)
+message(STATUS Configuring Kokkos-Tools)
+message(STATUS)
+
+# Tools settings
+set(KOKKOSTOOLS_ENABLE_CALIPER ON CACHE BOOL "Enable building Caliper library")
+set(KOKKOSTOOLS_ENABLE_APEX    ON CACHE BOOL "Enable building Apex library")
+
+# Check Kokkos settings
+if(TARGET Kokkos::kokkos)
+  message(STATUS "Using Kokkos settings for KokkosTools (source: ${Kokkos_SOURCE_DIR})")
+else()
+  message(WARNING "Not using Kokkos for tools configuration\n\tNote: using settings from installed is Kokkos not yet supported")
+endif()
+  # TODO: Configure based on Kokkos
+#       * check+test compile flags, especially ABI relevant
+#       * use installed version if Kokkos_ROOT or Kokkos_DIR is defined
+#       * check if Kokkos was already included in build with add_subdirectory()
+#       * error out otherwise: does it make sense to look into Kokkos
+#         source code without knowing it's *actual* build options ?
+
+#Turn MPI support off: (arbitrary or if Kokkos has MPI disabled)
+#CFLAGS += -DUSE_MPI=0
+
+# CXXFLAGS=-O3 -std=c++11 -g -shared -fPIC
+# -Wall -Wextra
+
+set(CMAKE_CXX_COMPILER ${Kokkos_CXX_COMPILER})
+set(CMAKE_CXX_STANDARD ${CMAKE_CXX_STANDARD_DEFAULT})
+
+# Libraries
+find_package(PAPI)
```

Comment:
There should be an option to enable/disable papi support

Reply:
This should (IMO) take the form of:

```cmake
option(KOKKOSTOOLS_USE_PAPI "Enable PAPI support" ON)
if(KOKKOSTOOLS_USE_PAPI)
    find_package(PAPI REQUIRED)
endif()
```

So I used to do what you did here in the past. Experience has taught me that it is _much_ better to be explicit: make it an option and fail if it is not found. Especially since you have Caliper and Apex subprojects -- users will likely miss that PAPI was not found bc there will be so many messages from adding those subprojects.

Reply:
Fixed, thank you (PAPI is now guarded with the option as suggested).